Runtime/Core/ApiGatewayManagement/AmazonGameLiftClientWrapper.cs

```diff
@@ -42,17 +56,143 @@ public async Task<SearchGameSessionsResponse> SearchGameSessions(SearchGameSessi
+        public async Task<string> RegisterCompute(string computeName, string fleetId, string fleetLocation, string ipAddress)
         {
-            return await _amazonGameLiftClient.DescribeGameSessionsAsync(request);
+            try
+            {
+                var registerComputeRequest = new RegisterComputeRequest()
+                {
+                    ComputeName = computeName,
+                    FleetId = fleetId,
+                    IpAddress = ipAddress,
+                    Location = fleetLocation
+                };
+                var registerComputeResponse = await RegisterCompute(registerComputeRequest);
+                
+                return registerComputeResponse.Compute.GameLiftServiceSdkEndpoint;
+            }
+            catch (Exception ex)
+            {
+                Console.WriteLine(ex.Message);
+                throw;
+            }
+        }
+
+        public async Task CreateCustomLocationIfNotExists(string fleetLocation)
+        {
+            try
+            {
+                var listLocationsResponse = await ListLocations(new ListLocationsRequest
+                {
+                    Filters = new List<string>{ "CUSTOM" }
+                });
+                
+                var foundLocation = listLocationsResponse.Locations.FirstOrDefault(l => l.LocationName.ToString() == fleetLocation);
+
+                if (foundLocation == null)
+                {
+                    var createLocationResponse = await CreateLocation(new CreateLocationRequest()
+                    {
+                        LocationName = fleetLocation
+                    });
+                    
+                    if (createLocationResponse.HttpStatusCode == HttpStatusCode.OK)
+                    {
+                        Console.WriteLine($"Created Custom Location {fleetLocation}");
+                    }
+                }
+            }
+            catch (Exception ex)
+            {
+                Console.WriteLine(ex.Message);
+            }
+        }
+        
+        public async Task<string> CreateFleet(ComputeType computeType, string fleetLocation)
+        {
+            try
+            {
+                var createFleetRequest = new CreateFleetRequest
+                {
+                    ComputeType = computeType,
+                    Description = FleetDescription,
+                    Locations = new List<LocationConfiguration>
+                    {
+                        new()
+                        {
+                            Location = fleetLocation
+                        }
+                    }
+                };
+                var createFleetResponse = await CreateFleet(createFleetRequest);
+                return createFleetResponse.FleetAttributes.FleetId;
+            }
+            catch (Exception ex)
+            {
+                Console.WriteLine(ex.Message);
+                throw;
```

Comment:
This is different from the error handling for the previous method. Is there a reason the previous silently fails but this doesn't?

Reply:
Most likely me just missing this, moving these to the UI and changing to UI errors should be cleaner and more informative to the user.